packages/lesswrong/server/resolvers/tagResolvers.ts

```diff
@@ -757,3 +760,79 @@ defineQuery({
     return { tags, totalCount };
   },
 });
+
+defineMutation({
+  name: "promoteLensToMain",
+  resultType: "Boolean",
+  argTypes: "(lensId: String!)",
+  fn: async (root, args: {lensId: string}, context) => {
+    const { lensId } = args;
+    if (!userIsAdminOrMod(context.currentUser)) {
+      throw new Error("Only admins can promote lenses to main");
+    }
+
+    // eslint-disable-next-line no-console
+    console.log(`Promoting lens ${lensId} to main`);
+    
+    // Find the lens and corresponding tag
+    const lensMultiDocument = await context.loaders.MultiDocuments.load(lensId);
+    if (!lensMultiDocument) throw new Error("Lens ID not found: "+lensId);
+    const tagId = lensMultiDocument.parentDocumentId;
+    const tag = await context.loaders.Tags.load(tagId);
+    if (!tag) throw new Error("Could not find corresponding tag for lens ID: "+lensId);
+    
+    // Swap revisions
+    const tagRevisions = await Revisions.find({ documentId: tagId }).fetch();
+    const lensRevisions = await Revisions.find({ documentId: lensId }).fetch();
```

Comment:
can be in a Promise.all

Reply:
True, but I really don't think that performance optimization is valuable here. This is a rare, admin-only mutation.